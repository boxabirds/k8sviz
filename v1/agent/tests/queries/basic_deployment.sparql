# Query: Create a basic web application deployment with 3 replicas
# This query retrieves the necessary information to generate a Deployment YAML

PREFIX k8s: <http://kubernetes.io/ontology#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?resource ?property ?value ?comment
WHERE {
  # Find Deployment resource
  ?deployment a k8s:Deployment ;
              rdfs:label ?deploymentLabel ;
              rdfs:comment ?deploymentComment .
  
  # Get deployment properties
  {
    SELECT ?property ?propertyLabel ?propertyComment
    WHERE {
      ?property rdfs:domain k8s:Deployment ;
                rdfs:label ?propertyLabel ;
                rdfs:comment ?propertyComment .
    }
  }
  
  # Get Pod template requirements
  UNION
  {
    SELECT ?pod ?podProperty ?podComment
    WHERE {
      k8s:Pod rdfs:comment ?podComment .
      ?podProperty rdfs:domain k8s:Pod ;
                   rdfs:label ?podPropertyLabel .
    }
  }
  
  # Get Container requirements
  UNION
  {
    SELECT ?container ?containerProperty ?containerComment
    WHERE {
      k8s:Container rdfs:comment ?containerComment .
      ?containerProperty rdfs:domain k8s:Container ;
                         rdfs:label ?containerPropertyLabel .
    }
  }
  
  # Get best practices
  UNION
  {
    SELECT ?practice ?practiceLabel ?practiceComment
    WHERE {
      ?practice a k8s:BestPractice ;
                rdfs:label ?practiceLabel ;
                rdfs:comment ?practiceComment .
      FILTER(CONTAINS(LCASE(STR(?practiceComment)), "deployment") || 
             CONTAINS(LCASE(STR(?practiceComment)), "workload"))
    }
  }
}