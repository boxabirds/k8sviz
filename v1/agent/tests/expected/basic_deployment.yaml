apiVersion: apps/v1
kind: Deployment
metadata:
  name: web-app
  labels:
    app: web-app
spec:
  replicas: 3
  selector:
    matchLabels:
      app: web-app
  template:
    metadata:
      labels:
        app: web-app
    spec:
      containers:
      - name: web-app
        image: nginx:latest
        ports:
        - containerPort: 80
        resources:
          requests:
            cpu: "100m"
            memory: "128Mi"
          limits:
            cpu: "500m"
            memory: "512Mi"
        livenessProbe:
          httpGet:
            path: /
            port: 80
          initialDelaySeconds: 30
          periodSeconds: 10
        readinessProbe:
          httpGet:
            path: /
            port: 80
          initialDelaySeconds: 5
          periodSeconds: 5